### ISSUE TITLE
Update contact picker in submit expense flow

### ISSUE THREAD

**Issue by dubielzyk-expensify** [Expensify/App#53894]
## Problem

From our recent usability studies and other observations we can see that people are struggling to submit to the right place.

## Solution

Given most people would wanna submit to a workspace let's make a seperate section at the top for the workspaces that the user can submit to and also show who the approver would be:

![image](https://github.com/user-attachments/assets/c770f68c-c6b7-4ab5-8929-d87db4187479)
![image](https://github.com/user-attachments/assets/acca51ae-2da2-425f-bdc4-baa954a71592)

Everything else remains the same

cc @Expensify/design @trjExpensify @JmillsExpensify

**Comment by dubielzyk-expensify** [comment]
It could be for `Create expense too`. I think we should probably merge the two mocks then?

**Comment by shawnborton** [comment]
Definitely agree there

**Comment by shawnborton** [comment]
I think the PR above does cover all the cases though, at least that's what it seemed like from running a quick test.

**Comment by trjExpensify** [comment]
Yeah, so maybe for a follow-up PR with this issue, we can switch out `Workspace` for the `submitsTo` on the workspace rows? 

<img width="222" alt="image" src="https://github.com/user-attachments/assets/47520993-8ee7-41a2-9a57-6b83c4ab9c8c" />

**Comment by dubielzyk-expensify** [comment]
Sounds good 👍 So which issue do we keep?

**Comment by trjExpensify** [comment]
We can keep the other linked to the PR. I've added that to the OP: 

<img width="854" alt="image" src="https://github.com/user-attachments/assets/4eeb8acd-fcaa-4267-8ec8-d6482b7524b0" />

**Comment by sakluger** [comment]
I'm going to add this to the `#convert` project since it has to do with new customer adoption. Let me know if it should live elsewhere.

**Comment by trjExpensify** [comment]
I think we can close this issue, per [here](https://github.com/Expensify/App/issues/53894#issuecomment-2540280977). Reopen if I missed something on that!
